tsl/src/continuous_aggs/common.c

```diff
@@ -239,16 +249,28 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 			Node *width_arg;
 			Node *col_arg;
 
-			if (!function_allowed_in_cagg_definition(fe->funcid))
+			/* Filter any non bucketing functions */
+			FuncInfo *finfo = ts_func_cache_get_bucketing_func(fe->funcid);
+			if (finfo == NULL || !finfo->is_bucketing_func)
+			{
 				continue;
+			}
+
+			/* Do we have a bucketing function that is not allowed in the CAgg definition */
+			if (!function_allowed_in_cagg_definition(fe->funcid))
+			{
+				if (IS_DEPRECATED_BUCKET_FUNC(finfo))
+				{
+					ereport(ERROR,
+							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							 errmsg("experimental bucket functions are not supported inside a CAgg "
+									"definition"),
+							 errhint("Use a function from the %s schema instead.",
+									 FUNCTIONS_SCHEMA_NAME)));
+				}
```

Comment:
How does it work during upgrade, will people be able to upgrade if they have such caggs?

Reply:
This function is only called when you create new CAggs. So, this PR blocks only the ability to create new CAggs using time_bucket_ng. Existing CAggs continue to work.

```sql

-- 2.14.2
CREATE MATERIALIZED VIEW conditions_summary_weekly
WITH (timescaledb.continuous, timescaledb.materialized_only=true) AS
SELECT city,
       timescaledb_experimental.time_bucket_ng('7 days', day, '2000-01-03' :: date) AS bucket,
       MIN(temperature),
       MAX(temperature)
FROM conditions
GROUP BY city, bucket
WITH NO DATA;

test1=# select * from _timescaledb_catalog.continuous_aggs_bucket_function ;
 mat_hypertable_id |                         bucket_func                         | bucket_width | bucket_origin | bucket_offset | bucket_timezone | bucket_fixed_width
-------------------+-------------------------------------------------------------+--------------+---------------+---------------+-----------------+--------------------
                 2 | timescaledb_experimental.time_bucket_ng(interval,date,date) | 604800000000 |               |               |                 | t
(1 row)


test1=# alter extension timescaledb update;
ALTER EXTENSION

INSERT into conditions values(now(), 'Hamburg', 34);
CALL refresh_continuous_aggregate('conditions_summary_weekly', NULL, NULL);

-- 2.15.0-dev
CREATE MATERIALIZED VIEW conditions_summary_weekly2
WITH (timescaledb.continuous, timescaledb.materialized_only=true) AS
SELECT city,
       timescaledb_experimental.time_bucket_ng('7 days', day, '2000-01-03' :: date) AS bucket,
       MIN(temperature),
       MAX(temperature)
FROM conditions
GROUP BY city, bucket
WITH NO DATA;

ERROR:  experimental bucket functions are not supported inside a CAgg definition
HINT:  Use a function from the _timescaledb_functions schema instead.
```